### PR TITLE
Add some testing/style guidelines to software page

### PR DESCRIPTION
**Pull request by Jason-Ku**
Adds some info about style and coding guidelines, some helpful info for testing as well. Onboarding/setup needs work though, will work on that

### CODE REVIEW

File: src/pages/teams/software.mdx [hunk 1]
```diff
@@ -1,28 +1,109 @@
...
+### General Guidelines
+
+- Try to keep unittests small and focused, they shouldn't be testing more the functionality of more than one method.
+- Use `assert` to verify any relevant conditions.
+- Make sure that unittests do not interact or depend on each other.
+
+### Mocking
+
+For the most part, unit tests involving API logic will require you to make use of the `mock` library to avoid issues with the database client.
+We don't want to be making actual calls to the database when we're testing Otherwiseher logic, after all.
```
Comment (Waidhoferj) [inline src/pages/teams/software.mdx]: "Otherwiseher"
Typo?

File: src/pages/teams/software.mdx [hunk 2]
```diff
@@ -1,28 +1,109 @@
...
+- Use `assert` to verify any relevant conditions.
+- Make sure that unittests do not interact or depend on each other.
+
+### Mocking
+
+For the most part, unit tests involving API logic will require you to make use of the `mock` library to avoid issues with the database client.
+We don't want to be making actual calls to the database when we're testing Otherwiseher logic, after all.	
+
+- What is mocking?
+  - "Mocking" is creating a special, dummy instance of an object or class that you, the developer, have complete control over.
```
Comment (Waidhoferj) [inline src/pages/teams/software.mdx:64]: Could you hyperlink the pyunit docs on mocking here?

**Review by Waidhoferj — CHANGES_REQUESTED** [review]
Nice work! It's cool to review this, since I get to learn about the role at the same time. Just a possible typo and a missing reference.

File: src/pages/teams/software.mdx (hunk 1 above)
Comment (Jason-Ku) [reply src/pages/teams/software.mdx]: Definite typo, thanks for pointing that out

File: src/pages/teams/software.mdx (hunk 2 above)
Comment (Jason-Ku) [reply src/pages/teams/software.mdx:64]: I have them linked a couple lines below, I think

File: src/pages/teams/software.mdx (hunk 1 above)
Comment (Jason-Ku) [reply src/pages/teams/software.mdx]: Fixed

**Review by Waidhoferj — APPROVED** [review]
Looks good

**Pull request merged**